### PR TITLE
Added nmake script for building ctest.dll and cpptest.dll

### PR DESCRIPTION
**Pull request by girish946**
Added nmake script for building ctest.dll and cpptest.dll

In `test/test.lua` added `test/clib` to `package.cpath`.
The is done in order to use `ctest` module with `require("ctest")`.

### CODE REVIEW

File: windows.yml [hunk 1]
```diff
@@ -10,5 +10,6 @@ steps:
 
   - script: |
      cd test
+     call test\msvctest-build.bat
```
Comment (fsfod) [inline windows.yml]: This should be nmake /F MAKEFILENAME Its silently failing  in [CI  tests](https://dev.azure.com/moonjit/moonjit/_build/results?buildId=235&view=logs&j=805631b4-8102-53fd-e75f-25423b543c0c&t=a0980f0c-d3df-5f8b-3308-b96e53d3b0e1&l=12) as path no found.

**Comment by fsfod** [comment]
The nmake file msvctest-build.bat should be renamed to something like Makefile.win since its not batch script file.

File: windows.yml (hunk 1 above)
Comment (girish946) [reply windows.yml]: done.

**Comment by girish946** [comment]
Thanks @fsfod for the review and also for providing Nmake code for windows test builds.


> The nmake file msvctest-build.bat should be renamed to something like Makefile.win since its not batch script file.

done.

**Comment by siddhesh** [comment]
Please add another Signed-off-by line in the commits for Thomas as well, you can get it from previous commits.

**Comment by fsfod** [comment]
it still failing will need to run vcvars bat so nmake is in the search path

**Comment by fsfod** [comment]
The first script block in the windows.yml file already setups nmake on the path so it would make more sense to move the test build commands there adding these two lines at the end of it.
```
cd ..\test
nmake /F Makefile.win
```

**Comment by girish946** [comment]
Having issues with this PR. Closing it for now.
Will send a new PR after fixing these issues.

**Pull request closed without merging**